frontend/src/components/CheckboxSelect.js

```diff
@@ -31,29 +31,37 @@ export function renderCheckboxes(
   });
 }
 
-export const makeCheckboxes = (options, checked) => (
-  options.reduce((obj, r) => ({ ...obj, [r.value]: checked }), {})
-);
-
-export default function CheckboxSelect(props) {
-  const {
-    options,
-    onApply,
-    labelId,
-    toggleAllText,
-    toggleAllInitial,
-    styleAsSelect,
-    labelText,
-    ariaName,
-    disabled,
-    hideToggleAll,
-  } = props;
+export const makeCheckboxes = (options, checked) => {
+  const checkboxes = options.reduce((obj, r) => ({ ...obj, [r.value]: checked }), {});
+  return checkboxes;
+};
 
+export default function CheckboxSelect({
+  options,
+  onApply,
+  labelId,
+  toggleAllText,
+  toggleAllInitial,
+  styleAsSelect,
+  labelText,
+  ariaName,
+  disabled,
+  hideToggleAll,
+  onChange,
+}) {
   const [toggleAllChecked, setToggleAllChecked] = useState(toggleAllInitial);
-  const [checkboxes, setCheckboxes] = useState(makeCheckboxes(options, toggleAllChecked));
+  const [checkboxes, setCheckboxes] = useState(
+    makeCheckboxes(options, toggleAllChecked),
+  );
 
   const menu = createRef();
 
+  useEffect(() => {
+    const checked = Object.keys(checkboxes).filter((checkbox) => checkboxes[checkbox]);
+    onChange(checked);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Are you running into issues adding `onChange` in the dependency list? This eslint rule is one I am really hesitant to ignore. It is probably the rule that has frustrated me the most, but ultimately ignoring it has always caused strange bugs and weird issues down the road. If you haven't already found it I found this post helpful https://github.com/facebook/react/issues/14920#issuecomment-471070149

Reply:
You're right to point this out, just the process of tracking down the re-renders led me to multiple bugs that needed fixing